chain/db/statedb.go

```diff
@@ -202,32 +288,98 @@ func (sdb *StateDB) deleteAccount(addr common.Uint160, acc *account) error {
 	return nil
 }
 
-func (sdb *StateDB) Finalise(deleteEmptyObjects bool) {
+func (sdb *StateDB) updateNanoPay(id string, nanoPay *nanoPay) error {
+	buff := bytes.NewBuffer(nil)
+	err := nanoPay.Serialize(buff)
+	if err != nil {
+		panic(fmt.Errorf("can't encode nano pay %v: %v", nanoPay, err))
+	}
+
+	return sdb.trie.TryUpdate(append(NanoPayPrefix, id...), buff.Bytes())
+}
+
+func (sdb *StateDB) deleteNanoPay(id string) error {
+	err := sdb.trie.TryDelete(append(NanoPayPrefix, id...))
+	if err != nil {
+		return err
+	}
+
+	delete(sdb.nanoPay, id)
+	return nil
+}
+
+func (sdb *StateDB) updateNanoPayCleanup(height uint32, nanoPayCleanup map[string]struct{}) error {
+	buff := bytes.NewBuffer(nil)
+
+	if err := serialization.WriteVarUint(buff, uint64(len(nanoPayCleanup))); err != nil {
+		panic(fmt.Errorf("can't encode nano pay cleanup %v: %v", nanoPayCleanup, err))
+	}
+	npcs := make([]string, 0)
```

Comment:
It's more efficient to use `make([]string, len(nanoPayCleanup))` here since map length is stored in Go

Reply:
fixed